File: C/약수의합.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    int answer = 0;
    double comp;
    
    for(int i = 1 ; i <= n ; i++){
        comp = ((double)n / i) - floor(((double)n / i));  
        if(comp == 0)  answer += i;
    }
    
    return answer;
}

File: python/약수의합.py
def solution(n):
    answer = 0
    
    for i in range(1,n+1):
        if not n % i:
            answer += i
            
    return answer

def solution(n):
    return n + sum([i for i in range(1, n // 2 + 1) if not n % i])